internal/engine/compiler/engine.go

```diff
@@ -702,14 +702,23 @@ func (ce *callEngine) Definition() api.FunctionDefinition {
 
 // Call implements the same method as documented on wasm.ModuleEngine.
 func (ce *callEngine) Call(ctx context.Context, params ...uint64) (results []uint64, err error) {
+	return ce.call(ctx, params, nil)
+}
+
+// CallWithStack implements the same method as documented on wasm.ModuleEngine.
+func (ce *callEngine) CallWithStack(ctx context.Context, stack []uint64) (res []uint64, err error) {
+	return ce.call(ctx, stack, stack[:0])
+}
+
+func (ce *callEngine) call(ctx context.Context, params []uint64, results []uint64) (_ []uint64, err error) {
```

Comment:
in the future, who knows.. it might pass the stack further down. For now, this is nice as it keeps the patch smaller. One nit:

```suggestion
func (ce *callEngine) call(ctx context.Context, params, results []uint64) (_ []uint64, err error) {
```